### DISCUSSION TITLE
Alpha 140

### DISCUSSION THREAD

**Discussion by Eugeny** [Eugeny/tabby#4012] [General]
* SSH: added experimental SFTP support
* SSH: added a File Transfers panel for both SFTP and Zmodem
* SSH: added PuTTY key format support
* SSH: fixed multiple Zmodem issues
* SSH: fixed session not getting disconnected when closing a tab
* SSH: added a WinSCP launch hotkey
* SSH: added support for multiple private keys per connection (useful when syncing config across platforms)
* Vault: added an encrypted container to store passwords and private keys when the OS keychain isn't available. The Vault is encrypted on disk and secured with a master passphrase, see [Vault Encryption](https://github.com/Eugeny/terminus/wiki/Vault-Encryption).
* Vault: the entire config file can be optionally encrypted and stored in the Vault
* Splitting a tab no longer restores entire history in the new pane
* Clicking a pane now cancels multi-pane input
* Windows: updated the bundled `clink` version
* Added CLI options to start SSH and serial connections
* Windows: auto-set Terminus as `%COMSPEC%` inside shells
* Serial: support for custom baud rates

<hr /><em>This discussion was created from the release <a href='https://github.com/Eugeny/terminus/releases/tag/v1.0.140'>Alpha 140</a>.</em>

**Comment by koalawangyang** [comment]
M1 MacBook Air can not run this version. Nothing display when the app launched on ARM 64 version.

**Reply by csimonmeunier** [reply]
Same for me on Intel MacBook Pro, on macOS Monterey beta

**Reply by kphk** [reply]
Same here

**Reply by Eugeny** [reply]
Monterey support is now added in the nightly

**Reply by ovidiutachegmail** [reply]
Same here. Crashing and not responding to any keystroke. I can't even quit it. I have to do it with Force quit.
<img width="912" alt="Screenshot 2021-06-22 at 22 07 26" src="https://user-images.githubusercontent.com/53333761/122985016-59d38b80-d3a6-11eb-8a20-e4b2491e9c14.png">

**Reply by wakkings** [reply]
Problem after update package.. 

Error: Can't resolve all parameters for SaveOutputContextMenu: ([object Object], ?, [object Object], [object Object]).

**Comment by SavuBogdan** [comment]
updated from 120 to 140 on windows and agent forwarding doesn't work anymore on any of my connections , had to put my key in to be able to connect and it's something not desirable for me since I have around 50 connections configured and having them tied to a file is troublesome at least , makes the config less portable.

**Reply by Eugeny** [reply]
Please try the nightly build - agent forwarding should be already fixed there.

**Reply by SavuBogdan** [reply]
Thanks for the response I have reconfigured all the connections with key in vault , ill probably wait until it's in a release .

I have a few questions about vault implementation , as i mentioned i have many connections but all of them use the same key i noticed i had to upload the key for each of them which seems unnecessary , maybe give us the options to add keys or passwords directly  in the vault and have them accessible  for all the connections instead of having the key linked to each connection even if it's the same file that way i can just update key in one place if i need to instead of wasting 2 hours to update in 50 places.

Also is the vault portable could i move it along with my config to a new laptop for example ? This is a major selling factor for me because it saves me much time when changing laptops.

**Reply by Eugeny** [reply]
Thanks for the feedback! I'll definitely add an option to pick existing keys. The vault is portable and you can use it across multiple machines, e.g. with the `sync-config` plugin.
